File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.py
from collections import Counter
import heapq

class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        num_frequencies = Counter(nums)
        frequency_arr = [[] for _ in range (len(nums) + 1)]

        for key,value in num_frequencies.items():
            frequency_arr[value].append(key)
        
        top_k = []
        for i in range (len(frequency_arr)-1, 0, -1):
            for num in frequency_arr[i]:
                top_k.append(num)
                if(len(top_k)) == k:
                    return top_k


        
        


        

File: Data Structures & Algorithms/remove-element/submission-1.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) 
    {
        int first = 0;
        int last = nums.size()-1;

        while(first <= last)
        {
            if(nums[first] == val)
            {
                nums[first] = nums[last];
                last--;
            }
            else
            {
                first++;
            }
        }

        return first;
    }
};